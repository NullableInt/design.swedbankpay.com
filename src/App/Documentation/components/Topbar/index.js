import React, { Component } from "react";
import PrismCode from "react-prism";

import { ComponentPreview, DocToc, Property, Attribute } from "#";
import TopbarComponent from "@/Topbar";
import { topbar } from "$/px-script/main";
import Alert from "@/Alert";

const menu = {
    btn: {
        icon: "menu",
        text: "Menu"
    },
    items: ["Link 1", "Link 2", "Link 3"]
};

const topbarContent = {
    id: "topbar-nav",
    ...menu
};

// TODO: Remove when deprecated parts are deleted.
const DeprecatedAlert = () => (
    <Alert type="danger">
        <h3>Component contains deprecated parts</h3>
        <p>The following parts are deprecated and will be removed:</p>
        <ul>
            <li>Right nav</li>
            <li>Levels in left nav</li>
            <li>topbar-btn-right and topbar-btn-left classes will be replaced by topbar-btn</li>
        </ul>
        <p>For continued support, please follow the examples in the documentation.</p>
    </Alert>
);

const Overview = () => (
    <>
        <h2 id="overview">Overview</h2>
        <p>
            To use a topbar the bare minimum markup required is:
        </p>
        <ul>
            <li><Property value=".topbar" /> wrapper.</li>
            <li>Anchor with class <Property value=".topbar-logo" /> to display the logo and make it clickable.</li>
        </ul>
        <p>
            With this you get a topbar with the PayEx logo.
            In the topbar you can add additional functionality like a menu containing links. To enable this add a <PrismCode className="language-html">{"<button>"}</PrismCode> and
            a <PrismCode className="language-html">{"<nav>"}</PrismCode> containing anchors. Remember to include <Attribute data name="toggle-nav" value="{your_nav_id}"/> as a <PrismCode className="language-html">{"<button>"}</PrismCode> attribute
            to get the menu to appear when the menu-button is clicked.
        </p>
        <ComponentPreview language="html" showCasePanel codeFigure>
            <TopbarComponent topbarContent={topbarContent} logout />
        </ComponentPreview>
    </>
);

const FixedTopbar = () => (
    <>
        <h2 id="fixed-topbar">Fixed topbar</h2>
        <p>
            Add <Property value=".topbar-fixed" /> with your <Property value=".topbar" /> to get a fixed topbar.
            For a demonstration of the usage see the documentations top bar or implement it in your own project!
        </p>
        <ComponentPreview language="html" codeFigure>
            <TopbarComponent fixed />
        </ComponentPreview>
    </>
);

const TopbarText = () => (
    <div className="col-lg-10 doc-body">
        <DeprecatedAlert />
        <p className="lead">Topbar...</p>
        <Overview />
        <FixedTopbar />
    </div>
);

class Topbar extends Component {
    componentDidMount () {
        topbar.init(1);
    }

    render () {
        return (
            <div className="doc-container">
                <div className="row">
                    <TopbarText />
                    <DocToc component={TopbarText} />
                </div>
            </div>
        );
    }
}

export default Topbar;

/* For testing */
export { Overview, TopbarText, FixedTopbar, DeprecatedAlert };
