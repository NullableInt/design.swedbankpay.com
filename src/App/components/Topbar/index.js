import React, { Fragment } from "react";
import PropTypes from "prop-types";

const TopbarBtn = ({ align, icon, text, target }) => (
    <button type="button" className={`topbar-btn-${align}`} data-toggle-nav={`#${target}`}>{"\n\t\t"}
        {icon ? <i className="material-icons topbar-btn-icon">{icon}</i> : null}{icon ? "\n\t\t" : null}
        {text ? <span className="topbar-btn-text">{text}</span> : null}{text ? "\n\t" : null}
    </button>
);

const LeftMenu = ({ align, menu }) => {
    const { id, hierarchy } = menu;

    const Slide = ({ slide }) => {
        const { id, items } = slide;

        const MenuItem = ({ item, last }) => {
            const { title, target, href } = item;

            return (
                <Fragment>
                    {href ? <a href={href} onClick={e => e.preventDefault()}>{title}</a> : <span data-target={`#nav-${align}-slide-${target}`}>{title}</span>}{!href || last ? "\n" : null}
                </Fragment>
            );
        };

        const GroupItem = ({ groupItem }) => {
            const { groupTitle, items } = groupItem;

            return (
                <div className="topbar-nav-group">
                    <div className="topbar-nav-group-heading">{groupTitle}</div>{"\n"}
                    {items.map((item, i) => <MenuItem key={i} item={item} last={i === items.length - 1} />)}
                </div>
            );
        };

        return (
            <div id={`nav-${align}-slide-${id}`} className={"topbar-nav-slide"}>
                <div className="slides-container">
                    {items.map((item, i) => (
                        <Fragment key={i}>{"\n"}
                            {item.groupTitle ? <GroupItem groupItem={item} /> : <MenuItem item={item} last={i === items.length - 1} />}
                        </Fragment>
                    ))}
                </div>
            </div>
        );
    };

    return (
        <nav id={id} className={`topbar-nav topbar-nav-${align}`}>
            {hierarchy.map(slide => <Slide key={slide.id} slide={slide}></Slide>)}
        </nav>
    );
};

const RightMenu = ({ align, menu }) => {
    const { id, items } = menu;

    return (
        <nav id={id} className={`topbar-nav topbar-nav-${align}`}>
            {items.map((item, i) => (
                <Fragment key={i}>{"\n"}
                    <a href={item.href} onClick={e => e.preventDefault()}>{item.title}</a>
                </Fragment>
            ))}{"\n"}
        </nav>
    );
};

const Topbar = ({ leftMenu, rightMenu }) => (
    <div className="topbar">{"\n"}
        {leftMenu ? <TopbarBtn align="left" icon={leftMenu.btn.icon} text={leftMenu.btn.text} target={leftMenu.id} /> : null}{leftMenu ? "\n" : null}
        <a href="#" className={"topbar-logo"} onClick={e => e.preventDefault()}></a>{"\n"}
        {rightMenu ? <TopbarBtn align="right" icon={rightMenu.btn.icon} text={rightMenu.btn.text} target={rightMenu.id} /> : null}{rightMenu ? "\n" : null}
        {leftMenu ? <LeftMenu align="left" menu={leftMenu} /> : null}{leftMenu ? "\n" : null}
        {rightMenu ? <RightMenu align="right" menu={rightMenu} /> : null}{rightMenu ? "\n" : null}
    </div>
);

Topbar.propTypes = {
    leftMenu: PropTypes.object,
    rightMenu: PropTypes.object
};

export default Topbar;
