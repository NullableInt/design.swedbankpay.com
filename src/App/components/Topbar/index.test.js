import React from "react";
import { shallow } from "enzyme";

import Topbar from "./index";

describe("Component: Topbar - ", () => {
    const menu = {
        btnLeft: {
            text: "Menu"
        },
        items: [
            {
                title: "Link 1",
                href: "#"
            },
            {
                title: "Link 2",
                href: "#"
            },
            {
                title: "Link 3",
                href: "#"
            }
        ]
    };

    const menuNoBtnIcon = {
        btn: {
            text: "Menu"
        },
        items: [
            {
                title: "Link 1",
                href: "#"
            },
            {
                title: "Link 2",
                href: "#"
            },
            {
                title: "Link 3",
                href: "#"
            }
        ]
    };

    const topbarContent = {
        id: "topbar-nav",
        ...menu
    };

    it("is defined", () => {
        expect(Topbar).toBeDefined();
    });

    it("renders", () => {
        const wrapper = shallow(<Topbar />);

        expect(wrapper).toMatchSnapshot();
    });

    it("renders with nav and button when prop topbarContent is provided", () => {
        const wrapper = shallow(<Topbar topbarContent={topbarContent} />);

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.html()).toContain("topbar-btn-left");
        expect(wrapper.html()).toContain("topbar-nav");
    });
});
