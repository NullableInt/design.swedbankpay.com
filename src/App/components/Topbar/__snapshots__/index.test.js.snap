// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: Topbar -  renders 1`] = `
<header
  className="topbar "
>
  

  <a
    className="topbar-logo"
    href="#"
    onClick={[Function]}
  />
  

</header>
`;

exports[`Component: Topbar -  renders a static topbar if prop fixed is true 1`] = `
<header
  className="topbar topbar-fixed"
>
  

  <a
    className="topbar-logo"
    href="#"
    onClick={[Function]}
  />
  

</header>
`;

exports[`Component: Topbar -  renders with a button right and a logo 1`] = `
<header
  className="topbar "
>
  

  <a
    className="topbar-logo"
    href="#"
    onClick={[Function]}
  />
  

  <TopbarBtn
    icon="exit_to_app"
    text="Log out"
  />
  

</header>
`;

exports[`Component: Topbar -  renders with nav and button left when prop topbarContent is provided 1`] = `
<header
  className="topbar "
>
  

  <TopbarBtn
    icon="menu"
    target="topbar-nav"
    text="Menu"
  />
  

  <TopbarMenu
    menu={
      Object {
        "btn": Object {
          "icon": "menu",
          "text": "Menu",
        },
        "id": "topbar-nav",
        "items": Array [
          "Link 1",
          "Link 2",
          "Link 3",
        ],
      }
    }
  />
  

  <a
    className="topbar-logo"
    href="#"
    onClick={[Function]}
  />
  

</header>
`;
